event/producer_test.go

```diff
@@ -84,3 +93,51 @@ func TestProducer_ContentUpdated(t *testing.T) {
 		})
 	})
 }
+
+func TestProducer_ReindexTaskCounts(t *testing.T) {
+	Convey("Given ReindexTaskCounts has been configured correctly", t, func() {
+		pChannels := &dpkafka.ProducerChannels{
+			Output: make(chan []byte, 1),
+		}
+
+		kafkaProducerMock := &kafkatest.IProducerMock{
+			ChannelsFunc: func() *dpkafka.ProducerChannels {
+				return pChannels
+			},
+		}
+
+		marshallerMock := &mock.MarshallerMock{
+			MarshalFunc: func(s interface{}) ([]byte, error) {
+				return schema.ReindexTaskCounts.Marshal(s)
+			},
+		}
+
+		// event is message
+		reindexTaskCountsProducer := event.ReindexTaskCountsProducer{
+			Producer:   kafkaProducerMock,
+			Marshaller: marshallerMock,
+		}
+
+		Convey("When ReindexTaskCounts is called on the event producer with EnablePublishReindexTaskCountsTopic enabled", func() {
+			err := reindexTaskCountsProducer.TaskCounts(ctx, &config.Config{EnableReindexTaskCounts: true}, expectedReindexTaskCountsEvent)
+			So(err, ShouldBeNil)
+
+			var avroBytes []byte
+			var testTimeout = time.Second * 5
+			select {
+			case avroBytes = <-pChannels.Output:
+				t.Log("avro byte sent to producer output")
+			case <-time.After(testTimeout):
+				t.Fatalf("failing test due to timing out after %v seconds", testTimeout)
+				t.FailNow()
```

Comment:
No need for FailNow() after calling Fatalf(...) as Fatalf is equivalent to Logf followed by FailNow

Reply:
Fixed.